From 9c128337b0e8f5ee06d19fdb8c876436d94952b4 Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Tue, 26 Jan 2021 16:02:41 +0200
Subject: [PATCH] [SW-30453] i2c indirect: remove ic from opcode 12

Opcode 12 - fw load, does not support the new definition of integrity
check and not going to. Removing the IC related code in order to remove
confusion.

Change-Id: Ia7677a774b76303b48c64a7e992939db4a62868f
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/preboot/i2cslave_indirect.c b/preboot/i2cslave_indirect.c
index 7277869..ab4d0a0 100644
--- a/preboot/i2cslave_indirect.c
+++ b/preboot/i2cslave_indirect.c
@@ -83,14 +83,10 @@
 	/*
 	 * Received firmware length is the last requset RDQL + previous
 	 * requests length which is calculate by the header offset.
-	 * If IC bit is set, the 4 last bytes of each request is not part
-	 * of the firmware but rather integrity check data, so we do not
-	 * consider them.
+	 * Integrity check is not supported for opcode 12 (load fw)
 	 */
 	unsigned int fw_len = req_wd.hdr.rqdl +
-		(req_wd.hdr.offset * HL_I2CI_REQ_MAX_RQDL) -
-		((req_wd.hdr.offset + 1) *
-		 req_wd.hdr.ic * 4);
+		(req_wd.hdr.offset * HL_I2CI_REQ_MAX_RQDL);
 
 	if (fit_validate_size(fw_len))
 		return; /* More data is needed */
@@ -215,13 +211,11 @@
 		 * The firmware data is being placed directly to FIT base
 		 * buffer, it may be assembled by multiple requests so we used
 		 * the offset int the header an index.
-		 * If IC bit is set, the 4 last bytes are not part of the
-		 * firmware but rather integrity check data, so we ignore and
-		 * override them.
+		 * Integrity check is not supported for opcode 12 (fw load)
 		 */
 		req_wd.length = size;
-		return (u8 *)FIT_IMG_BASE + (req_wd.hdr.offset *
-			(HL_I2CI_REQ_MAX_RQDL - req_wd.hdr.ic * 4));
+		return (u8 *)FIT_IMG_BASE +
+			(req_wd.hdr.offset * HL_I2CI_REQ_MAX_RQDL);
 
 	case HL_I2CI_REQ_OP_MEMCPY_CTRL:
 		upper_bound = offsetof(struct hl_i2ci_memcpy_ctrl, status);
